File: subcommands/daily.py
from rich.console import Console
from rich.markdown import Markdown
from rich.panel import Panel
from rich.table import Table
from rich.text import Text
import requests

console = Console()

def run():
    url = "https://leetcode-api-pied.vercel.app/daily"
    response = requests.get(url, timeout=5)
    data = response.json()
    
    title = data["question"]["title"]
    frontendId = data["question"]["questionFrontendId"]
    date = data["date"]
    difficulty = data["question"]["difficulty"]
    url = "https://leetcode.com" + data["link"]
    
    console.print(Panel.fit(
        f"[bold cyan]LeetCode Daily Challenge - {date}[/bold cyan]",
        title="📅 Daily Challenge"
    ))
    
    console.print(f"[bold yellow]{frontendId}. {title}[/bold yellow] ({difficulty})")
    console.print(f"[blue]{url}[/blue]\n")
    
    content = data["question"]["content"]
    md = Markdown(content, code_theme="monokai")
    console.print(md)

File: subcommands/daily.go
package subcommands

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"time"

	"github.com/CompProgTools/Kruskal/internal/models"
	"github.com/charmbracelet/lipgloss"
)

var (
	panelStyle = lipgloss.NewStyle().
		Border(lipgloss.RoundedBorder()).
		BorderForeground(lipgloss.Color("14")).
		Padding(1, 2)

	titleDaily = lipgloss.NewStyle().
		Bold(true).
		Foreground(lipgloss.Color("14"))
)

func RunDaily() error {
	url := "https://leetcode-api-pied.vercel.app/daily"

	client := &http.Client{Timeout: 5 * time.Second}
	resp, err := client.Get(url)
	if err != nil {
		return err
	}

	defer resp.Body.Close()

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		return err
	}

	var data models.LeetCodeDaily
	if err := json.Unmarshal(body, &data); err != nil {
		return err
	}

	title := data.Question.Title
	frontendID := data.Question.QuestionFrontentID
	date := data.Date
	difficulty := data.Question.Difficulty
	link := "https://leetcode/com" + data.Link

	fmt.Println(panelStyle.Render(titleDaily.Render(fmt.Sprintf("LeetCode Daily Challenge - %s", date))))
	fmt.Println()
	fmt.Printf("%s %s. %s (%s)\n", lipgloss.NewStyle().Foreground(lipgloss.Color("11")).Bold(true).Render("Problem:"), frontendID, title, difficulty)
	fmt.Printf("%s %s\n", lipgloss.NewStyle().Foreground(lipgloss.Color("12")).Render("Link:"), link)
	fmt.Println()
	fmt.Println(data.Question.Content)

	return nil
}